File: dbCreatorCli/src/main/java/com/github/fCat97/database/QueryCursor.java
package com.github.fCat97.database;

import media.uqab.fuzzybleJava.SqlCursor;
import java.sql.Connection;
import java.sql.ResultSet;
import java.sql.SQLException;
import java.sql.Statement;

public class QueryCursor implements SqlCursor {
    private final Connection mConnection;
    private final String mQuery;
    private Statement mStatement;
    private ResultSet resultSet;

    public QueryCursor(Connection connection, String query) {
        this.mConnection = connection;
        this.mQuery = query;
        try {
            this.mStatement = this.mConnection.createStatement();
            this.resultSet = this.mStatement.executeQuery(this.mQuery);
        } catch (SQLException e) {
            System.out.println(e.getMessage());
        }
    }

    @Override
    public String getString(int columnIndex) {
        try {
            return resultSet != null ? resultSet.getString(columnIndex + 1) : null;
        } catch (SQLException e) {
            System.out.println(e.getMessage());
            return null;
        }
    }

    @Override
    public boolean moveToNext() {
        try {
            return resultSet != null && resultSet.next();
        } catch (Exception e) {
            System.out.println(e.getMessage());
            return false;
        }
    }

    @Override
    public int count() {
        int count = 0;
        try {
            Statement st = mConnection.createStatement();
            ResultSet rs = st.executeQuery(mQuery);
            while (rs.next()) {
                count++;
            }
            rs.close();
            st.close();
            // System.out.println("count:" + count + " q:" + mQuery);
        } catch (Exception e) {
            System.out.println("count-error: " + e.getMessage());
        }
        return count;
    }

    @Override
    public void close() {
        try {
            if (resultSet != null) {
                resultSet.close();
            }
            mStatement.close();
        } catch (SQLException e) {
            System.out.println(e.getMessage());
        }
    }
}

File: dbCreator/src/jvmMain/kotlin/media/uqab/fuzzyble/dbCreator/database/QueryCursor.kt
package media.uqab.fuzzyble.dbCreator.database

import media.uqab.fuzzybleJava.SqlCursor
import java.sql.Connection
import java.sql.ResultSet
import java.sql.SQLException
import java.sql.Statement

class QueryCursor(private val mConnection: Connection, private val mQuery: String) : SqlCursor {
    private lateinit var mStatement: Statement
    private var resultSet: ResultSet? = null

    init {
        try {
            mStatement = mConnection.createStatement()
            resultSet = mStatement.executeQuery(mQuery)
        } catch (e: SQLException) {
            println(e.message)
        }
    }

    override fun getString(columnIndex: Int): String? {
        return try {
            resultSet?.getString(columnIndex + 1)
        } catch (e: SQLException) {
            println(e.message)
            null
        }
    }


    override fun moveToNext(): Boolean {
        return try {
            resultSet?.next() == true
        } catch (e: Exception) {
            println(e.message)
            false
        }
    }

    @Override
    public override fun count(): Int {
        var count = 0
        try {
            val st = mConnection.createStatement()
            val rs = st.executeQuery(mQuery)
            while (rs.next()) {
                count++
            }
            rs.close()
            st.close()
            println("count:$count q:$mQuery")
        } catch (e: Exception) {
            println("count-error: " + e.message)
        }

        return count
    }

    @Override
    override fun close() {
        try {
            resultSet?.close()
            mStatement.close()
        } catch (e: SQLException) {
            println(e.message)
        }
    }
}